src/emulator/functionsEmulatorRuntime.ts

```diff
@@ -246,10 +248,26 @@ async function resolveDeveloperNodeModule(
   return moduleResolution;
 }
 
+async function assertResolveDeveloperNodeModule(
+  frb: FunctionsRuntimeBundle,
+  name: string
+): Promise<SuccessfulModuleResolution> {
+  const resolution = await resolveDeveloperNodeModule(frb, name);
+  if (
+    !(resolution.installed && resolution.declared && resolution.resolution && resolution.version)
+  ) {
+    throw new Error(
+      `Assertion failure: could not fully resolve ${name}: ${JSON.stringify(resolution)}`
+    );
+  }
+
+  return resolution as SuccessfulModuleResolution;
```

Comment:
out of curiosity, if you've checked that all these things exist (above), is `as SuccessfulModuleResolution` required still? (I imagine it is if you still have it here, but that's interesting...)

Reply:
Yeah it is.  Something about type boolean and type true ?